src/gtp_path.erl

```diff
@@ -318,7 +325,12 @@ handle_event(cast,{handle_response, echo_request, _, _Msg}, State, Data) ->
     path_restart(undefined, peer_state(down, State), Data, []);
 
 handle_event(cast, down, State, Data) ->
-    path_restart(undefined, peer_state(down, State), Data, []);
+    case Data of
+        #{icmp_error_handling := ignore} ->
+            keep_state_and_data;
+        _ ->
+            path_restart(undefined, peer_state(down, State), Data, [])
+    end;
```

Comment:
this is wrong place to check for the flag. I needs to go close to the ICMP handling.

Reply:
just rechecked and gtp_path:down/2 is only used from the ICMP logic. Still, unconditionally use a test for ICMP on an interface that has a very generic name is not good.

So, I would suggest to add a `Source` field to the API, forward that value in cast, populate that with the value `icmp` in the socket module and combine the icmp_error check with a check for the icmp value.

Reply:
or alternatively, rename `down` to something else, e.g. `icmp_error`

Reply:
Please, re review.